### PR TITLE
CLI Quality - Lint - bodyclose

### PR DESCRIPTION
**Pull request by teresaromero**
# Proposed changes

Fixes https://okteto.atlassian.net/browse/LAKE-71

This Pr enables `bodyclose` as litern and resolves errors from it. 

## How to validate

Go to any client request and remove the closing of the response body or code a new request. Then run `make lint` to spot where the response hasnt closed the body.

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

**Review by jLopezbarb — APPROVED** [review]

**Review by AdrianPedriza — APPROVED** [review]

**Comment by andreafalzetti** [comment]
Moved to draft to avoid merging without checking deepsource first (or hear from @teresaromero if it's done) and fixing the merge conflict.

File: pkg/okteto/kubetoken_client.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func (c *kubeTokenClient) GetKubeToken(baseURL, namespace string) (types.KubeTok
 		return types.KubeTokenResponse{}, fmt.Errorf("failed to unmarshal kubetoken response: %w", err)
 	}
 
-	return kubeTokenResponse, nil
+	return kubeTokenResponse, resp.Body.Close()
```
Comment (jLopezbarb) [inline pkg/okteto/kubetoken_client.go:83]: Could we  move the `resp.Body.Close` to the line before and log it into the log file?

We are checking that `err != nil` from the calling function. If we get the response correctly but the body is not able to close we will be returning an error to the user that we could have ignore.

```suggestion
        defer func() {
          if err := resp.Body.Close(); err != nil {
            oktetoLog.Info("could not close the body: %s", err)
          }
        }
	return kubeTokenResponse, nil
```

File: pkg/okteto/kubetoken_client.go [hunk 2]
```diff
@@ -98,5 +98,5 @@ func (c *kubeTokenClient) CheckService(baseURL, namespace string) error {
 		return fmt.Errorf("CheckService %w: %s", errKubetokenNotAvailable, baseURL)
 	}
 
-	return nil
+	return resp.Body.Close()
```
Comment (jLopezbarb) [inline pkg/okteto/kubetoken_client.go:101]: Same as above. We should be ignoring the error but logging it in case we need that information

**Review by jLopezbarb — CHANGES_REQUESTED** [review]

File: pkg/okteto/kubetoken_client.go (hunk 2 above)
Comment (teresaromero) [reply pkg/okteto/kubetoken_client.go]: changed the code, let me know if 👍🏻

**Review by andreafalzetti — APPROVED** [review]

**Review by andreafalzetti — APPROVED** [review]

**Pull request merged**